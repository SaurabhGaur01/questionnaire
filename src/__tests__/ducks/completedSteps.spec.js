import {
    setStepAsCompleted,
    clearStepAsCompleted,
    completedStepsReducer,
} from '../../ducks/completedSteps';

const mockObject = {
    mockVariable: 'mockValue',
};

describe('completedSteps Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setStepAsCompleted should match the snapshot' , () => {
            expect(setStepAsCompleted(mockObject)).toMatchSnapshot();
        });
        it('clearStepAsCompleted should match the snapshot' , () => {
            expect(clearStepAsCompleted()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(completedStepsReducer()).toEqual({});
        });
        it('should return the passed state if called with no action' , () => {
            expect(completedStepsReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(completedStepsReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(completedStepsReducer('', setStepAsCompleted(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearStepAsCompleted' , () => {
            expect(completedStepsReducer(mockObject, clearStepAsCompleted())).toEqual({});
        });
    });
});