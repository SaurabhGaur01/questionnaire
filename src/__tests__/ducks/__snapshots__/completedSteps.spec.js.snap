// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`completedSteps Duck Tests Action Generator clearStepAsCompleted should match the snapshot 1`] = `
Object {
  "type": "modules/questionnaire-app/CLEAR_STEP_AS_COMPLETED",
}
`;

exports[`completedSteps Duck Tests Action Generator setStepAsCompleted should match the snapshot 1`] = `
Object {
  "data": Object {
    "mockVariable": "mockValue",
  },
  "type": "modules/questionnaire-app/SET_STEP_AS_COMPLETED",
}
`;
